Getting an item in the registry name causes the build to fail

### Issue tracker rule checks (read carefully)

- [x] I have checked that my problem <a href='https://github.com/MCreator/MCreator/issues?q=is%3Aissue' target='_blank'>is not already reported</a>
- [x] I have checked that my problem is not covered on <a href='https://mcreator.net/support/knowledgebase' target='_blank'>Knowledge Base</a> or on <a href='https://mcreator.net/wiki' target='_blank'>MCreator's Wiki</a>
- [x] I have made sure this bug is not plugin-related (tested with MCreator without plugins installed)
- [x] I have checked that my written content does not violate the <a href='https://mcreator.net/wiki/general-publishing-guidelines' target='_blank'>publishing guidelines</a>

### Issue description

When trying to get an item in the registry name, the build fails.

### How to reproduce this issue?

Step 1: Add any procedure containing a procedurel block for obtaining an item name in the registry
Result: Build failed

### Operating system

Windows

### Details

MCreator version: 2025.1
Generator version: 1.21.4-NeoForge

### Example workspace

[TestWorkspace.zip](https://github.com/user-attachments/files/18695535/TestWorkspace.zip)

### Logs

```Executing Gradle task: build 
Build info: MCreator 2025.1.99999, neoforge-1.21.4, 64-bit, 16224 MB, Windows 11, JVM 21.0.6, JAVA_HOME: C:\Users\curre\.jdks\corretto-21.0.6, started on: 2025-02-06-23:00:45 
 
Calculating task graph as no cached configuration is available for tasks: build 
> Task :jarJar NO-SOURCE 
> Task :processTestResources NO-SOURCE 
> Task :createMinecraftArtifacts UP-TO-DATE 
> Task :processResources 
> Task :compileJava 
[C:\Users\curre\MCreatorWorkspaces\testworkspace\src\main\java\net\mcreator\testworkspace\procedures\TestRegNameProcedure.java:18](file:///C:/Users/curre/MCreatorWorkspaces/testworkspace/src/main/java/net/mcreator/testworkspace/procedures/TestRegNameProcedure.java:18): error: no suitable constructor found for ItemStack(Optional  >) 
                _player.getInventory().armor.set(0, new ItemStack(BuiltInRegistries.ITEM.get(ResourceLocation.parse("minecraft:dirt")))); 
                                                    ^ 
    constructor ItemStack.ItemStack(ItemLike) is not applicable 
      (argument mismatch; Optional<Reference<Item>> cannot be converted to ItemLike) 
    constructor ItemStack.ItemStack(Holder<Item>) is not applicable 
      (argument mismatch; Optional<Reference<Item>> cannot be converted to Holder<Item>) 
    constructor ItemStack.ItemStack(Void) is not applicable 
      (argument mismatch; Optional<Reference<Item>> cannot be converted to Void) 
[C:\Users\curre\MCreatorWorkspaces\testworkspace\src\main\java\net\mcreator\testworkspace\procedures\TestRegNameProcedure.java:21](file:///C:/Users/curre/MCreatorWorkspaces/testworkspace/src/main/java/net/mcreator/testworkspace/procedures/TestRegNameProcedure.java:21): error: no suitable constructor found for ItemStack(Optional  >) 
                _living.setItemSlot(EquipmentSlot.FEET, new ItemStack(BuiltInRegistries.ITEM.get(ResourceLocation.parse("minecraft:dirt")))); 
                                                        ^ 
    constructor ItemStack.ItemStack(ItemLike) is not applicable 
      (argument mismatch; Optional<Reference<Item>> cannot be converted to ItemLike) 
    constructor ItemStack.ItemStack(Holder<Item>) is not applicable 
      (argument mismatch; Optional<Reference<Item>> cannot be converted to Holder<Item>) 
    constructor ItemStack.ItemStack(Void) is not applicable 
      (argument mismatch; Optional<Reference<Item>> cannot be converted to Void) 
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output 
2 errors 
> Task :compileJava FAILED 
FAILURE: Build failed with an exception. 
* What went wrong: 
Execution failed for task ':compileJava'. 
> Compilation failed; see the compiler output below. 
[C:\Users\curre\MCreatorWorkspaces\testworkspace\src\main\java\net\mcreator\testworkspace\procedures\TestRegNameProcedure.java:18](file:///C:/Users/curre/MCreatorWorkspaces/testworkspace/src/main/java/net/mcreator/testworkspace/procedures/TestRegNameProcedure.java:18): error: no suitable constructor found for ItemStack(Optional  >) 
                  _player.getInventory().armor.set(0, new ItemStack(BuiltInRegistries.ITEM.get(ResourceLocation.parse("minecraft:dirt")))); 
                                                      ^ 
      constructor ItemStack.ItemStack(ItemLike) is not applicable 
        (argument mismatch; Optional<Reference<Item>> cannot be converted to ItemLike) 
      constructor ItemStack.ItemStack(Holder<Item>) is not applicable 
        (argument mismatch; Optional<Reference<Item>> cannot be converted to Holder<Item>) 
      constructor ItemStack.ItemStack(Void) is not applicable 
        (argument mismatch; Optional<Reference<Item>> cannot be converted to Void) 
[C:\Users\curre\MCreatorWorkspaces\testworkspace\src\main\java\net\mcreator\testworkspace\procedures\TestRegNameProcedure.java:21](file:///C:/Users/curre/MCreatorWorkspaces/testworkspace/src/main/java/net/mcreator/testworkspace/procedures/TestRegNameProcedure.java:21): error: no suitable constructor found for ItemStack(Optional  >) 
                  _living.setItemSlot(EquipmentSlot.FEET, new ItemStack(BuiltInRegistries.ITEM.get(ResourceLocation.parse("minecraft:dirt")))); 
                                                          ^ 
      constructor ItemStack.ItemStack(ItemLike) is not applicable 
        (argument mismatch; Optional<Reference<Item>> cannot be converted to ItemLike) 
      constructor ItemStack.ItemStack(Holder<Item>) is not applicable 
        (argument mismatch; Optional<Reference<Item>> cannot be converted to Holder<Item>) 
      constructor ItemStack.ItemStack(Void) is not applicable 
        (argument mismatch; Optional<Reference<Item>> cannot be converted to Void) 
  Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output 
  2 errors 
* Try: 
> Check your code and dependencies to fix the compilation error(s) 
> Run with --scan to get full insights. 
BUILD FAILED in 6s 
3 actionable tasks: 2 executed, 1 up-to-date 
Configuration cache entry stored. 
 
BUILD FAILED 
Task completed in 7 seconds```

--- a/plugins/generator-1.21.4/neoforge-1.21.4/procedures/registryname_to_mcitem.java.ftl
+++ b/plugins/generator-1.21.4/neoforge-1.21.4/procedures/registryname_to_mcitem.java.ftl
@@ -1,2 +1,2 @@
 <#include "mcelements.ftl">
-BuiltInRegistries.ITEM.get(${toResourceLocation(input$registryname)})
+BuiltInRegistries.ITEM.getValue(${toResourceLocation(input$registryname)})

--- a/plugins/generator-1.21.4/neoforge-1.21.4/procedures/registryname_to_mcitemblock.java.ftl
+++ b/plugins/generator-1.21.4/neoforge-1.21.4/procedures/registryname_to_mcitemblock.java.ftl
@@ -1,2 +1,2 @@
 <#include "mcelements.ftl">
-BuiltInRegistries.BLOCK.get(${toResourceLocation(input$registryname)})
+BuiltInRegistries.BLOCK.getValue(${toResourceLocation(input$registryname)})

--- a/src/test/java/net/mcreator/integration/generator/GTProcedureBlocks.java
+++ b/src/test/java/net/mcreator/integration/generator/GTProcedureBlocks.java
@@ -148,6 +148,31 @@ public class GTProcedureBlocks {
 							</value></block>
 							""".formatted(testXML, testXML));
 					break;
+				case "MCItem":
+					procedure.procedurexml = wrapWithBaseTestXML("""
+							<block type="return_itemstack"><value name="return">%s</value></block>
+							""".formatted(testXML));
+					break;
+				case "MCItemBlock":
+					procedure.procedurexml = wrapWithBaseTestXML("""
+							<block type="return_blockstate"><value name="return">%s</value></block>
+							""".formatted(testXML));
+					break;
+				case "Direction":
+					procedure.procedurexml = wrapWithBaseTestXML("""
+							<block type="return_direction"><value name="return">%s</value></block>
+							""".formatted(testXML));
+					break;
+				case "Entity":
+					procedure.procedurexml = wrapWithBaseTestXML("""
+							<block type="return_entity"><value name="return">%s</value></block>
+							""".formatted(testXML));
+					break;
+				case "DamageSource":
+					procedure.procedurexml = wrapWithBaseTestXML("""
+							<block type="return_damagesource"><value name="return">%s</value></block>
+							""".formatted(testXML));
+					break;
 				case "ProjectileEntity": // Projectile blocks are tested with the "Shoot from entity" procedure
 					procedure.procedurexml = wrapWithBaseTestXML("""
 							<block type="projectile_shoot_from_entity">

